modules/programs/zig.nix

```diff
@@ -0,0 +1,47 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.zig;
+
+in {
+  meta.maintainers = [ maintainers.luisnquin ];
+
+  options = {
+    programs.zig = {
+      enable = mkEnableOption "zig";
+
+      package = mkOption pkgs {
+        type = types.package;
+        default = pkgs.zig;
+        defaultText = literalExpression "pkgs.zig";
+        description = "The Zig package to use.";
+      };
+
+      enableZshIntegration = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Zig’s Zsh integration.
+        '';
+      };
```

Comment:
```suggestion
      enableZshIntegration = mkEnableOption "Zig's Zsh integration";
```

Reply:
I'm barely new creating options, I just aligned myself to the pattern used in other options, [example here](https://github.com/nix-community/home-manager/blob/b84191db127c16a92cbdf7f7b9969d58bb456699/modules/programs/direnv.nix#L59).I'm barely new to creating options, I just aligned myself with the pattern used in other options, example here.

Reply:
I'm barely new to creating options, I just aligned myself with the pattern used in [other options](https://github.com/nix-community/home-manager/blob/b84191db127c16a92cbdf7f7b9969d58bb456699/modules/programs/direnv.nix#L59).